public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -659,18 +659,34 @@ export class LokiDatasource
 
   modifyQuery(query: LokiQuery, action: QueryFixAction): LokiQuery {
     let expression = query.expr ?? '';
+    // NB: Usually the labelKeys should be fetched and cached in the datasource,
+    // but there might be some edge cases where this wouldn't be the case.
+    // However the changed would make this method `async`.
+    const allLabels = this.languageProvider.getLabelKeys();
```

Comment:
Hmmm, I am worried that we will encounter the same issues as we did previously in log context -  when we were deciding if we should use the label in context or not. The issues was that there might be an actual label - e.g. `service` with values `service1, service2, service3`, but it is not part of the queried stream. And then log message was something like `time=1 test=2 service=A` and we had false positivity that the service is actual label, while it was parsed label for that case. 

I can't think of a way to fix this at the moment, but I also think that this solution won't hold long, before we get issues reported.

Reply:
Oh, right. I don't think we can solve that in Grafana, but I will raise it again with the Loki folks.